### ISSUE TITLE
[a11y]: Dropdown component usage and style documentation problems

### ISSUE THREAD

**Issue by mbgower** [carbon-design-system/carbon-website#3363]
This is a summary issue for various considerations in the dropdown documentation. The considerations are detailed in comments in a pdf export of each of the [Usage](https://ibm.box.com/s/q1ytxjunn1dhhyyk6nr7p8lpwg2iy2hw) and [Style](https://ibm.box.com/s/4ea8cbexk5vfme5rijvncn1xbmyntslf) pages. **They should be examined carefully.**

In summary, the high level problems are:
- the 'placeholder' option that populates an unopened dropdown is not in the list of options. This is unusual (and is causing problems with implementation)
- Carbon should consider breaking combo box into its own component (as it is in storybook)
- Carbon should explore positives and negatives to making dropdown and select variants of the same component.
- there are a lot of passages in the Usage text that need to be revisited
- consider only using Space to select/deselect checkboxes in the multiselect
- consider making a single option selected in a multiselect appear in the field (styled as a tag, with an X if necessary; although one could argue that is unnecesary and it _could_ just be a text value)
- focus and highlighting on the options list and options within the list is problematic (detailed in Style page)

Note: this issue on documentation is additional to the following issues opened against the _implementation_ of dropdown and its variants:

- https://github.com/carbon-design-system/carbon/issues/12720
- https://github.com/carbon-design-system/carbon/issues/12721
- https://github.com/carbon-design-system/carbon/issues/12722
- https://github.com/carbon-design-system/carbon/issues/12723

**Comment by sstrubberg** [comment]
Closing in favor of tracking via https://github.com/carbon-design-system/carbon/issues/13521
